python/lsst/daf/butler/registry/interfaces/_opaque.py

```diff
@@ -0,0 +1,198 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+"""Interfaces for the objects that manage opaque (logical) tables within a
+`Registry`.
+"""
+
+__all__ = ["OpaqueTableManager", "OpaqueTableRecords"]
+
+from abc import ABC, abstractmethod
+from typing import (
+    Any,
+    Iterator,
+    Optional,
+)
+
+from ...core.ddl import TableSpec
+from ._database import Database, StaticTablesContext
+
+
+class OpaqueTableRecords(ABC):
+    """An interface that manages the records associated with a particular
+    opaque table in a `Registry`.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of the opaque table.
+    """
+    def __init__(self, name: str):
+        self.name = name
+
+    @abstractmethod
+    def insert(self, *data: dict):
+        """Insert records into the table
+
+        Parameters
+        ----------
+        data
+            Each additional positional argument is a dictionary that represents
+            a single row to be added.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def fetch(self, **where: Any) -> Iterator[dict]:
+        """Retrieve records from an opaque table.
+
+        Parameters
+        ----------
+        where
```

Comment:
I think it's supposed to be `**where` here: https://developer.lsst.io/python/numpydoc.html#additional-positional-and-keyword-arguments, same for `delete()` below. And `*data` above.